### PR TITLE
Improve applyNegMuCorrection algorithm

### LINKED ISSUE #13201: Improve ApplyNegMuCorrection algorithm
separated from issue #13145 to progress testing for MuonicXRD GUI for MuonicXRD algorithm.

### PR DESCRIPTION
**Pull request by MattKing06**
Fixes #13201 

Rearrange the algorithm to be similar to other interface algorithms

### CODE REVIEW

File: Code/Mantid/Framework/PythonInterface/plugins/algorithms/ApplyNegMuCorrection.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    def combine(self,dataDir,runno,A2000,B2000,A3000,B3000,_OILE,_OIHE,spec):
+        if spec<10:
+            specNo='0'+str(spec)
+        else:
+            specNo=str(spec)
+        print dataDir+'ral0'+runno+'.rooth30'+specNo+'.dat'
+        #loading data
+        try:
+            ws3000=Load(Filename=dataDir+r'\ral0'+runno+'.rooth30'+specNo+'.dat', OutputWorkspace='ws3000')
+            print 'hello'
```
Comment (FedeMPouzols) [inline Code/Mantid/Framework/PythonInterface/plugins/algorithms/ApplyNegMuCorrection.py]: Is this print hello part of the algorithm? :)

File: Code/Mantid/Framework/PythonInterface/plugins/algorithms/ApplyNegMuCorrection.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            ws2000=Load(Filename=dataDir+r'\ral0'+runno+'.rooth20'+specNo+'.dat', OutputWorkspace='ws2000')
+        except RuntimeError:
+            print runno+' '+specNo+"not found"
```
Comment (FedeMPouzols) [inline Code/Mantid/Framework/PythonInterface/plugins/algorithms/ApplyNegMuCorrection.py]: I'd suggest adding amore informative message in this except.
Also, should it actually raise the exception? I understand that if this `Load()` fails the algorithm won't run correctly.

File: Code/Mantid/Framework/PythonInterface/plugins/algorithms/ApplyNegMuCorrection.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
+from mantid.api import *  # PythonAlgorithm, registerAlgorithm, WorkspaceProperty
+
+#pylint: disable=no-init, too-many-arguments
```
Comment (FedeMPouzols) [inline Code/Mantid/Framework/PythonInterface/plugins/algorithms/ApplyNegMuCorrection.py]: About the 'too-many-arguments' that we saw the other day, I think that this reasonable in this case. It just happens that the algorithm and some methods have many parameters.
An alternative is to create a data holder class, but then pylint will most likely complaint that your class has too many data members! So you would need to use a similar disable directive to allow it.

File: Code/Mantid/Framework/PythonInterface/plugins/algorithms/ApplyNegMuCorrection.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        RenameWorkspaces(ws_ral,Suffix=suf)
+        RenameWorkspaces(ws2000_corr,Suffix=suf)
+        RenameWorkspaces(ws3000_corr,Suffix=suf)
+
+        DeleteWorkspace(ws2000)
+        DeleteWorkspace(ws3000)
+        DeleteWorkspace(ws2000_corr_rebin)
+        DeleteWorkspace(ws3000_corr_rebin)
+
+        return
```
Comment (FedeMPouzols) [inline Code/Mantid/Framework/PythonInterface/plugins/algorithms/ApplyNegMuCorrection.py]: I think that this return is not required. It doesn't do any harm either.

File: Code/Mantid/Framework/PythonInterface/plugins/algorithms/ApplyNegMuCorrection.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        GIHE=self.getProperty("GainISISHighE").value
+        OIHE=self.getProperty("OffsetISISHighE").value
+        GILE=self.getProperty("GainISISLowE").value
+        OILE=self.getProperty("OffsetISISLowE").value
+
+        for run in range(first,last):
+            for spec in range(0,3):
+                runno=str(run)
+                self.combine(dataDir,runno,GRHE,ORHE,GIHE,OIHE,GILE,OILE,spec)
+                self.combine(dataDir,runno,GRHE,ORHE,GIHE,OIHE,GILE,OILE,10)
```
Comment (FedeMPouzols) [inline Code/Mantid/Framework/PythonInterface/plugins/algorithms/ApplyNegMuCorrection.py]: This is probably not doing what the original script seems to do.  Note that the second self.combine() is outside of the loop over range(0,3): https://github.com/mantidproject/mantid/issues/13145

**Comment by FedeMPouzols** [comment]
@MattKing06 : the algorithm looks decently organized to me. Please have a look at the comments that I wrote inline. Maybe the documentation of the parameters is a bit sparse, but if that's what the scientists have provided for now we can go ahead with it and try to add a bit of explanation later on.

I see that you've split the original issue (#13145) into a few (doc, unit test, etc.). But in the original issue there was also this `Get_NegMu_MuonicXRD` algorithm in addition to `ApplyNegMuCorrection`. Have you added another issue for that one?

**Comment by MattKing06** [comment]
Yes, the GetNegMuMuonicXRD algorithm itself can be found at #13155 however you it also has a custom dialog that can be found at #13146

**Comment by MattKing06** [comment]
Agreed, I will change it to raise the exception.

**Pull request merged**